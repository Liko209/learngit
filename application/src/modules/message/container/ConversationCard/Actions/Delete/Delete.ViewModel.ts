/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2018-12-09 10:10:02
 * Copyright Â© RingCentral. All rights reserved.
 */

import { computed } from 'mobx';
import { ENTITY_NAME } from '@/store';
import { getEntity } from '@/store/utils';
import { StoreViewModel } from '@/store/ViewModel';
import { PostService } from 'sdk/module/post';
import { Post } from 'sdk/module/post/entity';
import PostModel from '@/store/models/Post';
import { Props, ViewProps } from './types';
import { ServiceLoader, ServiceConfig } from 'sdk/module/serviceLoader';
import { catchError } from '@/common/catchError';

class DeleteViewModel extends StoreViewModel<Props> implements ViewProps {
  private _postService: PostService;

  constructor(props: Props) {
    super(props);
    this._postService = ServiceLoader.getInstance<PostService>(
      ServiceConfig.POST_SERVICE,
    );
  }

  @computed
  get id() {
    return this.props.id;
  }

  @computed
  get disabled() {
    return this.props.disabled;
  }

  @computed
  get post() {
    return getEntity<Post, PostModel>(ENTITY_NAME.POST, this.id);
  }

  @catchError.flash({
    server: 'message.prompt.deletePostFailedForServerIssue',
    network: 'message.prompt.deletePostFailedForNetworkIssue',
  })
  deletePost = async () => {
    await this._postService.deletePost(this.id);
  }
}

export { DeleteViewModel };
