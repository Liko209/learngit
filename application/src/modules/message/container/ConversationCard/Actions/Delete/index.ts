/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2018-12-09 10:10:02
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './Delete';
