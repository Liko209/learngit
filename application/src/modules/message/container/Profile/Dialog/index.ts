/*
 * @Author: Devin Lin (devin.lin@ringcentral.com)
 * @Date: 2018-12-05 10:29:28
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './Group';
export * from './Person';
