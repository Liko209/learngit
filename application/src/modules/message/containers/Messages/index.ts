/*
 * @Author: Devin Lin (devin.lin@ringcentral.com)
 * @Date: 2018-10-08 18:38:33
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './Messages';
