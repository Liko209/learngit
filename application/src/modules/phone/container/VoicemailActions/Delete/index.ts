/*
 * @Author: joy.zhang (joy.zhang@ringcentral.com)
 * @Date: 2019-05-24 07:28:07
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './Delete';
