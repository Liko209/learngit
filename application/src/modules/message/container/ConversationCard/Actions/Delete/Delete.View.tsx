/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2018-12-09 10:10:02
 * Copyright Â© RingCentral. All rights reserved.
 */
import * as React from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { observer } from 'mobx-react';
import { JuiMenuItem } from 'jui/components';
import { Dialog } from '@/containers/Dialog';
import { ViewProps } from './types';
import { mainLogger } from 'sdk';

type Props = ViewProps & WithTranslation;

@observer
class DeleteViewComponent extends React.Component<Props> {
  private _handleDelete = () => {
    const { deletePost, t } = this.props;
    Dialog.confirm({
      modalProps: { 'data-test-automation-id': 'deleteConfirmDialog' },
      okBtnProps: { 'data-test-automation-id': 'deleteOkButton' },
      cancelBtnProps: { 'data-test-automation-id': 'deleteCancelButton' },
      title: t('message.prompt.deletePostTitle'),
      content: t('message.prompt.deletePostContent'),
      okText: t('common.dialog.delete'),
      okType: 'negative',
      cancelText: t('common.dialog.cancel'),
      onOK() {
        deletePost().catch((e: any) => {
          mainLogger.error(e);
        });
      },
    });
  };
  render() {
    const { disabled, t } = this.props;
    return (
      <JuiMenuItem
        onClick={this._handleDelete}
        disabled={disabled}
        icon="delete"
        data-test-automation-id="message-action-delete"
      >
        {t('message.action.deletePost')}
      </JuiMenuItem>
    );
  }
}

const DeleteView = withTranslation()(DeleteViewComponent);

export { DeleteView };
