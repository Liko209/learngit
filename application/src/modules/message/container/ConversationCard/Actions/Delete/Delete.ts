/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2018-12-09 10:10:02
 * Copyright Â© RingCentral. All rights reserved.
 */

import { buildContainer } from '@/base';
import { DeleteView } from './Delete.View';
import { DeleteViewModel } from './Delete.ViewModel';
import { Props } from './types';

const Delete = buildContainer<Props>({
  View: DeleteView,
  ViewModel: DeleteViewModel,
});

export { Delete, Props };
