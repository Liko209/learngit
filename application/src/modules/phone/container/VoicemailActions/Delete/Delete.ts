/*
 * @Author: joy.zhang (joy.zhang@ringcentral.com)
 * @Date: 2019-05-24 07:28:07
 * Copyright Â© RingCentral. All rights reserved.
 */

import { buildContainer } from '@/base';
import { DeleteView } from './Delete.View';
import { DeleteViewModel } from './Delete.ViewModel';
import { DeleteProps } from './types';

const Delete = buildContainer<DeleteProps>({
  View: DeleteView,
  ViewModel: DeleteViewModel,
});

export { Delete };
