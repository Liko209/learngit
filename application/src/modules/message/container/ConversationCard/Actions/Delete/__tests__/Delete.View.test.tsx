/*
 * @Author: Shining Miao (shining.miao@ringcentral.com)
 * @Date: 2019-08-26 19:14:53
 * Copyright Â© RingCentral. All rights reserved.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { DeleteView } from '../Delete.View';
import { Dialog } from '@/containers/Dialog';

describe('DeleteView', () => {
  describe('render()', () => {
    it('should render correct delete post dialog. [JPT-473]', () => {
      jest.spyOn(Dialog, 'confirm').mockImplementation(jest.fn());
      const props: any = {
        t: (str: string) => str,
        deletePost: () => {},
        disabled: false,
      };

      const wrapper = shallow(<DeleteView {...props} />);
      wrapper.simulate('click');

      expect(JSON.stringify(Dialog.confirm.mock.calls[0][0])).toMatchSnapshot();
    });
  });
});
