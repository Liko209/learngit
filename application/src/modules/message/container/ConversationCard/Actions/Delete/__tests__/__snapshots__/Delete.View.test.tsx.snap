// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteView render() should render correct delete post dialog. [JPT-473] 1`] = `"{\\"modalProps\\":{\\"data-test-automation-id\\":\\"deleteConfirmDialog\\"},\\"okBtnProps\\":{\\"data-test-automation-id\\":\\"deleteOkButton\\"},\\"cancelBtnProps\\":{\\"data-test-automation-id\\":\\"deleteCancelButton\\"},\\"title\\":\\"message.prompt.deletePostTitle\\",\\"content\\":\\"message.prompt.deletePostContent\\",\\"okText\\":\\"common.dialog.delete\\",\\"okType\\":\\"negative\\",\\"cancelText\\":\\"common.dialog.cancel\\"}"`;
